ldj-readme-renderer/render.js

```diff
@@ -8,45 +8,57 @@
 const fs = require('node:fs');
 const process = require('node:process');
 
-const OUTPUT = `README.md`;
+const OUTPUT = `../README.md`;
 
-const TEMPLATE = `[![Coverage](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=coverage)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Technical Debt](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_index)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Reliability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=reliability_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-# Living Documentation Java
-
-This project is an extension of the [Living Documentation][ldoc] set of tools to
-Java. As part of the Living Documentation toolchain, it can be used to generate
-documentation and diagrams from source code so that documentation is always
-up-to-date.
-
-## Analyzer
-
-The analyzer parses Java projects into Living Documentation JSON files that can
-be rendered by Living Documentation renderers. It accepts the following command
-line options:
+const json = JSON.parse(fs.readFileSync(process.argv[2]));
 
+// Section for rendering the command line options
+const CMD_LINE_TABLE_TEMPLATE = `
 | Short option | Long option | Description |
 | ------------ | ----------- | ----------- |
 ~
-
-[ldoc]: https://github.com/eNeRGy164/LivingDocumentation
 `;
 
-function render(d) {
+function renderCmdOptions(d) {
   if (d.Name === 'addOption' || d.Name === 'addRequiredOption') {
     const args = d.Arguments.map(a => a.Text).map(JSON.parse);
     const short = args[0] ? `\`-${args[0]}\`` : '';
     const long = args[1] ? `\`--${args[1]}\`` : '';
     const required = d.Name.includes('Required') ? '(Required.)' : '';
     return `| ${short} | ${long} | ${args[3]} ${required} |\n`;
   } else if (d instanceof Array) {
-    return d.map(render).join('');
+    return d.map(renderCmdOptions).join('');
   } else if (d instanceof Object) {
-    return Object.values(d).map(render).join('');
+    return Object.values(d).map(renderCmdOptions).join('');
   }
 }
 
-const json = JSON.parse(fs.readFileSync(process.argv[2]));
-fs.writeFileSync(OUTPUT, TEMPLATE.replace('~', () => render(json)));
+const cmdOptionsTable = CMD_LINE_TABLE_TEMPLATE.replace('~', () => renderCmdOptions(json));
+
+// Section rendering the implemented visited methods
+
+const JAVAPARSER_DOC_SITE = 'https://javadoc.io/static/com.github.javaparser/javaparser-core/3.25.6/';
+function getMarkdownLink(type) {
+  let splitted = type.split('.');
```

Comment:
Here and below, all except `finalReadme` can be `const` instead of `let`.

Reply:
I am not that familiar with the JS code styling guidelines, so excuse me for my ignorance (:  But why would `const` be preferred if it does not semantically represent a constant?
And if using `const`, would you also not need to change the naming to capital snake case?

Reply:
We didn't officially pick any JavaScript style guide, but if you want to be consistent with the existing code, I went with:

* mark all variables `const` where possible, to indicate we don't expect them to be reassigned
* give only "real" constants (in this case, hard-coded settings) shouty-case names

That happens to align with Google JavaScript style, see [here](https://google.github.io/styleguide/jsguide.html#features-use-const-and-let) and [here](https://google.github.io/styleguide/jsguide.html#naming-constant-names).

Reply:
I have changed this, for the report, it may be also nice to mention this style guide in the QA